File: 1493-longest-subarray-of-1s-after-deleting-one-element/1493-longest-subarray-of-1s-after-deleting-one-element.cpp
class Solution {
public:
    int longestSubarray(vector<int>& nums) {
        int n = nums.size();
        int left = 0,zeros = 0, ans = 0;
        for(int right=0;right<n;right++){
            if(nums[right]==0)
                zeros++;
            while(zeros>1){
                if(nums[left]==0){
                    zeros--;
                }
                left++;
            }
            ans = max(ans,right-left);
        }
        return ans;
    }
};

File: 1493-longest-subarray-of-1s-after-deleting-one-element/1493-longest-subarray-of-1s-after-deleting-one-element.py
class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        left,ans,zeros=0,0,0
        for right in range(len(nums)):
            if nums[right]==0:
                zeros += 1
            while zeros > 1:
                if nums[left] == 0:
                    zeros -= 1
                left += 1
            ans = max(ans, right - left)
        return ans